embassy-nrf/src/saadc.rs

```diff
@@ -263,9 +268,19 @@ impl<'d, const N: usize> Saadc<'d, N> {
 
     /// Continuous sampling with double buffers.
     ///
+    /// NOTE: It is important that the time spent within the callback supplied
```

Comment:
DMA overwriting the buffer that the user is using is undefined behavior, we can't have this in a safe API.

Reply:
Actually I see now that we are updating the next buffer to write to on EVENTS _STARTED. We need to update it post this event AND the callback having finished.

Reply:
I will push another commit shortly for this additional logic. My goal is to absolutely prevent undefined behavior and retain the safe API.

Reply:
Commit c36337531238175018a531b2dfe1951417056d9b should take care of things.

Reply:
> Actually I see now that we are updating the next buffer to write to on EVENTS _STARTED. We need to update it post this event AND the callback having finished.

I was wrong about this. Double buffering logic has been restored. Commit fa3a0439740f4d06f07343a1645f54723431ebd7 takes care of this.